superduperdb/ext/llm/training.py

```diff
@@ -178,14 +181,70 @@ class LLMTrainingArguments(TrainingArguments):
     max_seq_length: int = 512
     setup_chat_format: bool = False
     log_to_db: bool = False
+    ray_configs: t.Optional[t.Dict] = None
+    on_ray: bool = False
 
-    def __post_init__(self):
-        ...
-        # Overwrite __post_init__ for lazy build
-        # Cause we can run on remote ray, that can avoid building error on client side
+    def __post_init__(self, artifacts):
+        assert not self.output_dir
+        self.output_dir = f'{self.identifier}'
+        return super().__post_init__(artifacts)
 
     def build(self):
-        super().__post_init__()
+        self.__post_init__()
+
+    @staticmethod
+    def get_compute_metrics(metrics):
+        if not metrics:
+            return None
+
+        def compute_metrics(eval_preds):
+            output = {}
+            logits, labels = eval_preds
+            for metric in metrics:
+                output[metric.identifier] = metric(logits, labels)
+            return output
+
+        return compute_metrics
+
+    def fit(
+        self,
+        model: _Fittable,
+        db: Datalayer,
+        train_dataset: t.Union[QueryDataset, NativeDataset],
+        valid_dataset: QueryDataset,
+    ):
+        if isinstance(train_dataset, QueryDataset):
+            train_dataset = NativeDataset.from_list(
+                list(train_dataset)  # type: ignore[call-overload]
+            )
+
+        eval_datasets = {}
+        for vs in model.validation_sets:
+            unpacked = [r.unpack() for r in vs.data]
+            eval_datasets[vs.identifier] = NativeDataset.from_list(unpacked)
+        if not eval_datasets:
+            if isinstance(valid_dataset, QueryDataset):
+                valid_dataset = NativeDataset.from_list(
+                    list(valid_dataset)  # type: ignore[call-overload]
+                )
+            eval_datasets['_DEFAULT'] = valid_dataset
```

Comment:
```
        eval_datasets = {}
         for vs in model.validation_sets:
             unpacked = [r.unpack() for r in vs.data]
             eval_datasets[vs.identifier] = NativeDataset.from_list(unpacked)
         if not eval_datasets:
             if isinstance(valid_dataset, QueryDataset):
                 valid_dataset = NativeDataset.from_list(
                     list(valid_dataset)  # type: ignore[call-overload]
                 )
             eval_datasets['_DEFAULT'] = valid_dataset

    else:
            eval_datasets = valid_dataset
```

If there is only the input verification set, remove the key, otherwise all metrics will have `_DEFAULT` prefix